### ISSUE TITLE
Dockerfile

### ISSUE THREAD

**Issue by Taha-Chaudhry** [RayzrDev/Gameboi#8]
Could you make a Dockerfile for the bot? It would ultimately just make it easier to run, as well as make the bot easier to distribute.

**Comment by Taha-Chaudhry** [comment]
Here's a Dockerfile for anyone who needs it that works okay-ish (not optimised for maximum performance)

```
FROM openjdk:9
ARG JAR_FILE=target/*.jar
COPY ${JAR_FILE} app.jar
ENTRYPOINT ["java","-jar","/app.jar"]
```

**Comment by Taha-Chaudhry** [comment]
Ok I just realised that you can't edit the config.yml inside the container, so here is the fix for that:


This line will copy the config.yml from the docker container and paste it in your current directory

`docker cp NAME_OF_CONTAINER:/config.yml .`

Change the config.yml however you want, and make sure to add your actual token inside too



Afterwards, in the same path that you wrote the previous command in, write this line:

`docker cp config.yml NAME_OF_CONTAINER:/config.yml`

This line will copy the config.yml in your current path, and paste it into your docker container, removing the old config.yml along with it.


And of course, make sure to substitute `NAME_OF_CONTAINER` in both above commands for the name of the docker container you have your bot running in.

Hopefully you benefited from this :^)

**Comment by foxfirecodes** [comment]
thanks for this! i have created a proper dockerfile with a multi-stage build here: #9 

it has a few caveats, it's a little clunky to deal with the config.yml which you seem to be familiar with here, but I think there's a better way to go about it, just don't have the time to look right now

please let me know if this works better for you!

**Comment by Taha-Chaudhry** [comment]
alright I seemed to have created a workaround which fixes the config.yml for you. Obviously this isn't my main focus right now, which is why I took a while to do this, but I should be able to put the full Dockerfile here in a few days.

**Comment by foxfirecodes** [comment]
closing due to inactivity, I don't super actively maintain the bot anyway but I might include a proper Dockerfile in the future. for now I've merged #9 in.

**Comment by Taha-Chaudhry** [comment]
Actually forgot about this repo, but took a look at #9 anyway. I was unable to find the Dockerfile on Docker Hub, but saw this: “The only problem as it stands is that the bot runs as a root, meaning your data folder will be owned by root and so will your config.yml. one easy solution to this is to run the bot the first time to generate data/config.yml, and then run this to fix perms:
```bash
sudo chown -R $(whoami):$(whoami) data
```

A fix for this is to add this after your ‘FROM’ and before running your commands:
```bash
RUN addgroup -S bot && adduser -S bot -G bot
USER bot:bot
```

This gives your host more security by allowing the container to run without root perms, preventing back doors into the host system from the container.

**Comment by foxfirecodes** [comment]
thx! I just merged #13 in with the improvements you mentioned and pushed to Docker Hub. FYI using -G and -S gave me errors about ambiguous arguments, I know from personal experience that used to work but maybe `adduser` & `addgroup` are more strict now. either way, I tweaked it and it worked great.

(i assumed `-S` meant `--system` but could see no reason why I would want or need to have that flag enabled, so I omitted it)
